lib/visitors_manipulator.rb

```diff
@@ -24,17 +24,22 @@ def update
     go_forward = visit.visitors.all?(&:valid?) && visit.valid?(:visitors_set)
 
     if params[:next] == 'Add another visitor'
-      if visit.visitors.size < Visit::MAX_VISITORS
-        visit.visitors << model_class.new
-        redirect_to this_path
-      else
-        redirect_to this_path, notice: "You may only have a maximum of #{Visit::MAX_VISITORS} visitors"
-      end
+      max_visitors
     else
       redirect_to go_forward ? next_path : this_path
     end
   end
 
+  def max_visitors
+    if visit.visitors.size < Visit::MAX_VISITORS
+      visit.visitors << model_class.new
+      redirect_to this_path
+    else
+      flash[:notice] = I18n.t(:max_visitors, scope: 'controllers.shared')
+      redirect_to this_path
+    end
+  end
```

Comment:
@threedaymonk yeah, you re right and by moving redirect after the `if` then we won't have duplications.